packages/roosterjs-editor-dom/lib/selection/experimentalAttributeBasedSelectionMarker.ts

```diff
@@ -0,0 +1,137 @@
+import Position from './Position';
+import createRange from './createRange';
+import queryElements from '../utils/queryElements';
+
+const SELECTION_START_OFFSET_ATTRIBUTE = 'data-offset1';
+const SELECTION_END_OFFSET_ATTRIBUTE = 'data-offset2';
+const SELECTION_START_SELECTOR = `[${SELECTION_START_OFFSET_ATTRIBUTE}]`;
+const SELECTION_END_SELECTOR = `[${SELECTION_END_OFFSET_ATTRIBUTE}]`;
+
+type SerializablePosition = {
```

Comment:
RFC name suggestions

This type is different from a Selection in that 
- it guarantees it's anchorElement is an Element, not a Node. This means that the anchor will always have an HTML tag that we can assign attributes to.
- offsetCharacters can step into child text nodes.